### DISCUSSION TITLE
Cypress binary is missing

### DISCUSSION THREAD

**Discussion by Sowgam** [cypress-io/cypress#28366] [Questions and Help]
> npx npm-force-resolutions

added 238 packages in 23s

51 packages are looking for funding
  run `npm fund` for details
Installing Cypress (version: 13.5.1)

[STARTED] Task without title.
The cypress npm package is installed, but the Cypress binary is missing.

We expected the binary to be installed here: /home/jenkins/.cache/Cypress/13.5.1/Cypress/Cypress

Reasons it may be missing:

- You're caching 'node_modules' but are not caching this path: /home/jenkins/.cache/Cypress
- You ran 'npm install' at an earlier build step but did not persist: /home/jenkins/.cache/Cypress

**Comment by portyu9** [comment]
Hi @Sowgam 

The npm package is present, but the separate Cypress application binary was never installed or was not preserved between Jenkins stages.

The expected binary is:

```text
/home/jenkins/.cache/Cypress/13.5.1/Cypress/Cypress
```

Start by checking the Jenkins environment:

```bash
npx cypress cache path
npx cypress cache list
```

If `13.5.1` is missing, install it explicitly:

```bash
npx cypress install
npx cypress verify
```

In CI, the usual pattern is:

```bash
npm ci
npx cypress verify
npx cypress run
```

and persist/cache:

```text
/home/jenkins/.cache/Cypress
```

between relevant build stages.

Do **not** solve this by caching only `node_modules`. The `cypress` npm module and Cypress binary live in different places. Cypress's current CI guidance specifically recommends caching the global Cypress binary cache and your package manager's cache rather than directly caching `node_modules`.

Also check whether your install command uses:

```bash
--ignore-scripts
```

or:

```text
CYPRESS_INSTALL_BINARY=0
```

because either can intentionally prevent the normal Cypress `postinstall` binary download.

If lifecycle scripts are blocked by policy, that is fine — just make the explicit:

```bash
npx cypress install
```

step part of the Jenkins pipeline.

Please mark this answer as accepted if it helped, thank you.
